File: TagService/src/main/java/io/duryskuba/interestmatcher/TagService/resource/NotificationDTO.java
package io.duryskuba.interestmatcher.TagService.resource;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Collection;
import java.util.Set;


@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class NotificationDTO {

    //private PostDTO post;

    private Long objectId;
    private NotificationType type;
    private String content;
    private String author;

    private Long subscriber;
    private Collection<Tag> tags;

//    public static NotificationDTO of(PostDTO postDTO, Long subscriber, Collection<Tag> tags) {
//        return new NotificationDTO(postDTO, subscriber, tags);
//    }

}

File: NotificationService/src/main/kotlin/io/duryskuba/interstmatcher/NotificationService/resource/NotificationDTO.kt
package io.duryskuba.interstmatcher.NotificationService.resource


data class NotificationDTO (
    val id: Long? = 0,
    val objectId: Long? = 0,
    val type: NotificationType? = null,
    val url: String? = "",
    val content: String? = "",
    val author: String? = "",
    val subscriber: Long? = 0,
    val tags: Collection<Tag>? = ArrayList(),
    val status: NotificationStatus? = null
)